tests/build.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+set -exuo pipefail
+
+# This script basically builds bootc from source using the provided base image,
+# then runs the target tests.
+
+mkdir -p /tmp/tmp-bootc-build
+BOOTC_TEMPDIR="/tmp/tmp-bootc-build"
+
+# Get OS info from TEST_OS env
+OS_ID=$(echo "$TEST_OS" | cut -d '-' -f 1)
+OS_VERSION_ID=$(echo "$TEST_OS" | cut -d '-' -f 2)
+
+# Base image
+case "$OS_ID" in
+    "centos")
+        TIER1_IMAGE_URL="quay.io/centos-bootc/centos-bootc:stream${OS_VERSION_ID}"
+        ;;
+    "fedora")
+        TIER1_IMAGE_URL="quay.io/fedora/fedora-bootc:${OS_VERSION_ID}"
+        ;;
+esac
+
+CONTAINERFILE="${BOOTC_TEMPDIR}/Containerfile"
```

Comment:
This would create a *third* copy of our Dockerfile, I think we can avoid that by just having the github action matrix be *directly* passed down into e.g.

```
base=$1
shift
podman build --build-arg=base=$base
```

And that's all right?